go/app/server.go

```diff
@@ -180,6 +279,55 @@ func parseGetImageRequest(r *http.Request) (*GetImageRequest, error) {
 	return req, nil
 }
 
+type GetItemToIdResponse struct {
+	Name string `json:"name"`
+	Category string `json:"category"`
+	Image_Name string `json:"image_name"`
+}
+
+func (s *Handlers) GetItemToId(w http.ResponseWriter, r *http.Request) {
+	// get itemId from URL
+	uri := strings.Split(r.URL.Path, "/")
+	if len(uri) < 3 {
+		http.Error(w, "invalid request", http.StatusBadRequest)
+		return
+	}
+	itemId, _ := strconv.Atoi(uri[2])
+
+	// open json file
+	fileName := s.itemRepo.GetFileName()
+	jsonFile, err := os.Open(fileName)
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+	defer jsonFile.Close()
+
+	// read json file as bytes
+	bytes, _ := io.ReadAll(jsonFile)
+
+	// decode bytes to map
+	var decodeData map[string][]Item
+	json.Unmarshal(bytes, &decodeData)
+
+	var items []Item
+	for _, v := range decodeData {
+		items = append(items, v...)
+	}
+
+	// find item by itemId
+	resp := GetItemToIdResponse{
+		Name: items[itemId].Name,
+		Category: items[itemId].Category,
+		Image_Name: items[itemId].Image_Name,
+	}
```

Comment:
Please check the length of `items` is less than `itemId` before reference `items[itemId]` like:
```go
if len(items) < itemId {
    ... throw invalid item id error ...
}
```
and `itemId - 1` may be correct ?
```go
resp := GetItemToIdResponse{
    Name: items[itemId - 1].Name,
    Category: items[itemId - 1].Category,
    Image_Name: items[itemId - 1].Image_Name,
}
```